karoo_gp/base_class.py

```diff
@@ -1787,1535 +2043,1588 @@ class labels provided in the data .csv. Outputs an array of tuples
+              f'{branch_depth} < 0')
+        # # consider special instructions for this (pause) - 2019 06/08
+        # self.fx_karoo_pause()
+
+    # the point of mutation ('branch_top') chosen resides at the maximum
+    # allowable depth, so mutate term to term
+    elif branch_depth == 0:
+
+        log(f'\t\033[36m max depth branch node\033[1m '
+            f'{tree.root[3][branch_top]} \033[0;0m\033[36mmutates '
+            f'from \033[0;0m\033[36mmutates from \033[1mterm\033[0;0m '
+            '\033[36mto \033[1mterm\033[0;0m\n', display=['i'])
+        log(f'\n\n\033[33m *** Grow Mutation: terminal to terminal '
+            f'*** \033[0;0m\n\n\033[36m This is the unaltered '
+            f'tourn_winner:\033[0;0m\n {tree.root}', display=['db'])
 
-        # replaces 2 instances, below; tested 2016 07/09
-        branch_top = int(branch[0])
-        # 'tree_depth_max' - depth at 'branch_top' to set max potential
-        # size of new branch - 2016 07/10
-        branch_depth = self.tree_depth_max - int(tree[4][branch_top])
-
-        if branch_depth < 0:  # this has never occured ... yet
-            print('\n\t\033[31m ERROR! In fx_evolve_grow_mutate: branch_depth',
-                  branch_depth, '< 0')
-            # consider special instructions for this (pause) - 2019 06/08
-            self.fx_karoo_pause()
-
-        # the point of mutation ('branch_top') chosen resides at the maximum
-        # allowable depth, so mutate term to term
-        elif branch_depth == 0:
-
-            self.log('\t\033[36m max depth branch node\033[1m '
-                     f'{tree[3][branch_top]} \033[0;0m\033[36mmutates '
-                     'from \033[0;0m\033[36mmutates from \033[1mterm\033[0;0m '
-                     '\033[36mto \033[1mterm\033[0;0m\n', display=['i'])
-            self.log('\n\n\033[33m *** Grow Mutation: terminal to terminal '
-                     '*** \033[0;0m\n\n\033[36m This is the unaltered '
-                     f'tourn_winner:\033[0;0m\n {tree}', display=['db'])
+        # call the previously loaded .csv which contains all terminals
+        rnd = rng.integers(0, len(terminals) - 1)
+        tree.root[6][branch_top] = terminals[rnd]  # replace terminal (variable)
+
+        log(f'\n\033[36m This is tourn_winner after terminal\033[1m '
+            f'{branch_top} \033[0;0m\033[36mmutation, branch deletion, '
+            f'and updates:\033[0;0m\n {tree.root}', display=['db'])
+        pause(display=['db'])
+
+    # the point of mutation ('branch_top') chosen is at least
+    # one depth from the maximum allowed
+    else:
+
+        # TEST & DEBUG: force to 'func' or 'term' and comment the next 3 lines
+        # type_mod = '[func or term]'
+        rnd = rng.integers(2)
+        if rnd == 0:
+            type_mod = 'func'  # randomly selected as Function
+        elif rnd == 1:
+            type_mod = 'term'  # randomly selected as Terminal
```

Comment:
```suggestion
        type_mod = rng.choice(['term', 'func'])
```
If you want to be consistent with the previous code, maybe they should be `'Terminal'` and `'Function'`.